Orange/widgets/visualize/owradviz.py

```diff
@@ -365,66 +379,38 @@ def error(err):
 
         super().check_data()
         if self.data is not None:
-            if len(self.data) < 2:
-                error(self.Error.no_instances)
-            elif len(self.primitive_variables) < 3:
+            if len(self.primitive_variables) < 3:
                 error(self.Error.no_features)
+            else:
+                dom = self.data.domain
+                all_vars = [v for v in chain(dom.variables,
+                                             dom.metas) if v.is_primitive()]
+                if len(self.primitive_variables) < len(all_vars):
+                    self.Warning.removed_vars()
 
     def init_attr_values(self):
         super().init_attr_values()
         self.selected_vars = []
 
-    def get_embedding(self):
-        self.valid_data = None
-        if self.data is None:
-            return None
-
-        self.Warning.no_features.clear()
-        if len(self.model_selected) < 2:
-            self.Warning.no_features()
-            return None
-
-        ec, proj, msk = radviz(self.data, self.model_selected, self.projection)
-        angle = np.arctan2(*proj.T[::-1])
-        self.projection = np.vstack((np.cos(angle), np.sin(angle))).T
-        self.valid_data = msk
-
-        self.Warning.invalid_embedding.clear()
-        if ec is None or np.any(np.isnan(ec)):
-            self.Warning.invalid_embedding()
-            return None
-
-        embedding = np.full((len(self.data), 2), np.nan)
-        embedding[self.valid_data] = ec
-        return embedding
-
-    def get_anchors(self):
-        if self.projection is None:
-            return None, None
-        return self.projection, [a.name for a in self.model_selected]
-
     def _manual_move(self, anchor_idx, x, y):
         angle = np.arctan2(y, x)
         super()._manual_move(anchor_idx, np.cos(angle), np.sin(angle))
 
-    def send_components(self):
-        components = None
-        if self.data is not None and self.valid_data is not None and \
-                self.projection is not None:
-            angle = np.arctan2(*self.projection.T[::-1])
-            meta_attrs = [StringVariable(name='component')]
-            components = Table(Domain(self.model_selected, metas=meta_attrs),
-                               np.row_stack((self.projection.T, angle)),
-                               metas=np.array([["RX"], ["RY"], ["angle"]]))
-            components.name = self.data.name
-        self.Outputs.components.send(components)
+    def _send_components_x(self):
+        components_ = super()._send_components_x()
+        angle = np.arctan2(*components_[::-1])
+        return np.row_stack((components_, angle))
+
+    def _send_components_metas(self):
+        return np.vstack((super()._send_components_metas(), ["angle"]))
 
     def clear(self):
         if self.model_selected:
             self.model_selected.clear()
         if self.model_other:
             self.model_other.clear()
         super().clear()
+        self.projector = RadViz()
```

Comment:
I'd rather leave it as it is, since things are not as simple in OWFreeViz. The `projector` is not necessarily fitted, so the `projection` is somehow artificially obtained. Beside that the constructor gets anchors as a parameter.